From 31433dfecd72dd034abeaea0b76e3a994a23a363 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:24 -0400
Subject: [ethernet] igb: Add ethtool offline tests for i354

Message-id: <1409062359-13063-7-git-send-email-sassmann@redhat.com>
Patchwork-id: 89126
O-Subject: [RHEL7.1 PATCH 06/81] igb: Add ethtool offline tests for i354
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Fujinaka, Todd" <todd.fujinaka@intel.com>

Add the ethtool offline tests for i354 devices.

Signed-off-by: Todd Fujinaka <todd.fujinaka@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a4e979a27db3eb77e286dbe484e96c0c9c986e83)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index 86d5142..caa78d5 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -1659,7 +1659,8 @@ static int igb_setup_loopback_test(struct igb_adapter *adapter)
   if ((hw->device_id == E1000_DEV_ID_DH89XXCC_SGMII) ||
   (hw->device_id == E1000_DEV_ID_DH89XXCC_SERDES) ||
   (hw->device_id == E1000_DEV_ID_DH89XXCC_BACKPLANE) ||
-  (hw->device_id == E1000_DEV_ID_DH89XXCC_SFP)) {
+  (hw->device_id == E1000_DEV_ID_DH89XXCC_SFP) ||
+  (hw->device_id == E1000_DEV_ID_I354_SGMII)) {
 
    /* Enable DH89xxCC MPHY for near end loopback */
    reg = rd32(E1000_MPHY_ADDR_CTL);
@@ -1725,7 +1726,8 @@ static void igb_loopback_cleanup(struct igb_adapter *adapter)
  if ((hw->device_id == E1000_DEV_ID_DH89XXCC_SGMII) ||
  (hw->device_id == E1000_DEV_ID_DH89XXCC_SERDES) ||
  (hw->device_id == E1000_DEV_ID_DH89XXCC_BACKPLANE) ||
- (hw->device_id == E1000_DEV_ID_DH89XXCC_SFP)) {
+ (hw->device_id == E1000_DEV_ID_DH89XXCC_SFP) ||
+ (hw->device_id == E1000_DEV_ID_I354_SGMII)) {
   u32 reg;
 
   /* Disable near end loopback on DH89xxCC */
-- 
1.7.1